cmp/grid/impl/GridPersistenceModel.js

```diff
@@ -5,100 +5,76 @@
  * Copyright © 2020 Extremely Heavy Industries Inc.
  */
 import {HoistModel, XH} from '@xh/hoist/core';
-import {debounced} from '@xh/hoist/utils/js';
-import {cloneDeep, find, isUndefined, isEmpty, omit} from 'lodash';
+import {observable, action} from '@xh/hoist/mobx';
+import {find, isUndefined, omit} from 'lodash';
+import {PersistenceProvider, LocalStorageProvider} from '@xh/hoist/persistence';
 
 /**
- * Model for serializing/de-serializing saved grid state across user browsing sessions
- * and applying saved state when initialized by its parent `GridModel`.
+ * Model to manage persisting state from GridModel.
  *
- * GridModels can enable persistent grid state via their `stateModel` config, typically
- * provided as a simple string `gridId` to identify a given grid instance.
- *
- * It is not necessary to manually create instances of this class within an application.
  * @private
  */
 @HoistModel
-export class GridStateModel {
-
-    /**
-     * Version of grid state definitions currently supported by this model.
-     * Increment *only* when we need to abandon all existing grid state that might be saved on
-     * user workstations to ensure compatibility with a new serialization or approach.
-     */
-    static GRID_STATE_VERSION = 1;
+export class GridPersistenceModel {
 
-    /** @member {GridModel} */
+    VERSION = 1;  // Increment to abandon state.
     gridModel;
-    /** @member {string} */
-    gridId;
-    state = {};
-
-    /**
-     * @param {Object} c - GridStateModel configuration.
-     * @param {string} c.gridId - unique identifier for a Grid instance.
-     * @param {boolean} [c.trackColumns] - true to save state of columns,
-     *      including visibility, ordering and pixel widths.
-     * @param {boolean} [c.trackSort] - true to save sorting.
-     * @param {boolean} [c.trackGrouping] - true to save column grouping.
-     */
-    constructor({gridId, trackColumns = true, trackSort = true, trackGrouping = true}) {
-        this.gridId = gridId;
-        this.trackColumns = trackColumns;
-        this.trackSort = trackSort;
-        this.trackGrouping = trackGrouping;
-    }
-
-    /**
-     * Attach this object to a GridModel and have it observe / apply changes to tracked properties.
-     * @param {GridModel} gridModel
-     */
-    init(gridModel) {
+    path;
+
+    @observable.ref
+    state;
+
+    constructor(
```

Comment:
We should document what these arguments are

Reply:
I would agree - even for the private class, it's nice to have the ctor and its args documented and helps I think with additional IDE type inference when looking at / working on the class.